information_schema.go

```diff
@@ -0,0 +1,174 @@
+	&sql.Column{Name: "last_update_time", Type: sql.Blob, Source: FilesTableName, Nullable: true},
+	&sql.Column{Name: "last_access_time", Type: sql.Blob, Source: FilesTableName, Nullable: true},
+	&sql.Column{Name: "recover_time", Type: sql.Blob, Source: FilesTableName, Nullable: true},
+	&sql.Column{Name: "transaction_counter", Type: sql.Blob, Source: FilesTableName, Nullable: true},
+	&sql.Column{Name: "version", Type: sql.Int64, Source: FilesTableName, Nullable: true},
+	&sql.Column{Name: "row_format", Type: sql.Text, Source: FilesTableName, Nullable: true},
+	&sql.Column{Name: "table_rows", Type: sql.Blob, Source: FilesTableName, Nullable: true},
+	&sql.Column{Name: "avg_row_length", Type: sql.Blob, Source: FilesTableName, Nullable: true},
+	&sql.Column{Name: "data_length", Type: sql.Blob, Source: FilesTableName, Nullable: true},
+	&sql.Column{Name: "max_data_length", Type: sql.Blob, Source: FilesTableName, Nullable: true},
+	&sql.Column{Name: "index_length", Type: sql.Blob, Source: FilesTableName, Nullable: true},
+	&sql.Column{Name: "data_free", Type: sql.Int64, Source: FilesTableName, Nullable: true},
+	&sql.Column{Name: "create_time", Type: sql.Blob, Source: FilesTableName, Nullable: true},
+	&sql.Column{Name: "update_time", Type: sql.Blob, Source: FilesTableName, Nullable: true},
+	&sql.Column{Name: "check_time", Type: sql.Blob, Source: FilesTableName, Nullable: true},
+	&sql.Column{Name: "checksum", Type: sql.Blob, Source: FilesTableName, Nullable: true},
+	&sql.Column{Name: "status", Type: sql.Text, Source: FilesTableName, Nullable: true},
+	&sql.Column{Name: "extra", Type: sql.Blob, Source: FilesTableName, Nullable: true},
+}
+
+func newColumnStatisticsTable() sql.Table {
+	return &emptyTable{
+		name:   ColumnStatisticsTableName,
+		schema: columnStatisticsSchema,
+	}
+}
+
+var columnStatisticsSchema = sql.Schema{
+	&sql.Column{Name: "schema_name", Type: sql.Text, Source: ColumnStatisticsTableName},
+	&sql.Column{Name: "table_name", Type: sql.Text, Source: ColumnStatisticsTableName},
+	&sql.Column{Name: "column_name", Type: sql.Text, Source: ColumnStatisticsTableName},
+	&sql.Column{Name: "histogram", Type: sql.JSON, Source: ColumnStatisticsTableName},
+}
+
+type emptyTable struct {
+	name   string
+	schema sql.Schema
+}
+
+var _ sql.Table = (*emptyTable)(nil)
+
+// Name implements the sql.Table interface.
+func (t *emptyTable) Name() string { return t.name }
+
+// String implements the sql.Table interface.
+func (t *emptyTable) String() string { return printTable(t.name, t.schema) }
+
+// Schema implements the sql.Table interface.
+func (t *emptyTable) Schema() sql.Schema { return t.schema }
+
+// Partitions implements the sql.Table interface.
+func (t *emptyTable) Partitions(_ *sql.Context) (sql.PartitionIter, error) {
+	return new(partitionIter), nil
+}
+
+// PartitionRows implements the sql.Table interface.
+func (t *emptyTable) PartitionRows(_ *sql.Context, _ sql.Partition) (sql.RowIter, error) {
+	return new(rowIter), nil
```

Comment:
Instead of this, delete `rowIter` and use `sql.RowsToRowIter()`, which returns an empty iterator